src/ws-dropdown/dropdown-menu-item.js

```diff
@@ -96,40 +108,44 @@ export class DropdownMenuItem extends Component {
    */
   onClick = event => {
     event.stopPropagation();
+    // We prevented blur with mouse down event to ensure the click is captured before the dropdown closed
+    if ('activeElement' in document) {
+      document.activeElement.blur();
+    }
+    const {item} = this.state;
     // Do nothing if this item is disabled
-    if (this.state.disabled) {
+    if (item.disabled) {
       return;
     }
     // Click on parent means back navigation
     if (this.props.isParent) {
       // dropdown-item(go-back) -> dropdown-menu(show-parent) -> dropdown-item(show-parent) -> dropdown-menu
       this.props.handle('go-back');
     // Show next menu if item has children
-    } else if (this.state.children && this.state.children.length) {
+    } else if (item.children && item.children.length) {
       this.props.handle('show-child', this.menu);
-    } else {
-      if (!this.context.multiple) {
-        // If it is selected we publish null because it will be deselected in the upper menu
-        if (this.state.selected) {
-          this.props.handle('change', null);
-        } else {
-          this.setState({
-            selected : true,
-            stored : true,
-          });
-
-          this.props.handle('change', this.state);
-        }
+    } else if (!this.context.multiple) {
+      // If it is selected we publish null because it will be deselected in the upper menu
+      if (item.selected) {
+        this.props.handle('change', null);
       } else {
-        this.setState({selected : !this.state.selected});
+        item.selected = true;
+        item.stored = true;
+
+        this.setState({item});
```

Comment:
shouldn't this be with a callback and using prevState? Basically what you do here is
`this.setState({item: this.state.item})` as you reference it at the top, right ?

Reply:
Yes, the item is still referenced and I just call the setState fn to tell Preact/React to re-render.
What is the benefit from calling setState with a callback?

Reply:
I read about this asynchonous problem with setState but here it is in my eyes definitely not a problem. They say that you should not rely on this.state after calling setState and I definitely don't use this.state from the beginning of this function until the beginning of the render function, which indeed is called by setState. So no need to introduce more complex code here.